src/components/Button/index.tsx

```diff
@@ -0,0 +1,168 @@
+import type { MouseEventHandler, ReactElement } from 'react'
+import { hexToCSSFilter } from 'hex-to-css-filter'
+import styled from '@emotion/styled'
+import type { Theme } from '@emotion/react'
+
+const ICON_SIZE = {
+  LARGE: 24,
+  SMALL: 16
+}
+export const BUTTON_STYLE_KEYS = {
+  GHOST: 'ghost',
+  OUTLINE: 'outline',
+  OUTLINE_DISABLED: 'outlineDisabled',
+  SOLID_DISABLED: 'solidDisabled',
+  SOLID_PRIMARY: 'solidPrimary',
+  SOLID_SUB: 'solidSub'
+}
+
+type ButtonStyle = typeof BUTTON_STYLE_KEYS[keyof typeof BUTTON_STYLE_KEYS]
+type ButtonSize = 'large' | 'medium' | 'small'
+export interface ButtonProps {
+  buttonStyle?: ButtonStyle
+  size?: ButtonSize
+  type?: 'button' | 'submit' | 'reset'
+  children: string
+  option?: string
+  iconUrl?: string
+  onClick: MouseEventHandler<HTMLButtonElement>
+}
+type StyledButtonProps = Pick<ButtonProps, 'buttonStyle' | 'size'>
```

Comment:
```suggestion
interface StyledButtonProps {
  buttonStyle: ButtonStyle
  size: ButtonSize
}
```
어떤 값이 기본값인지 명시적으로 알 수 있게 기본값을 컴포넌트 props쪽으로 이동시키고 style에 대한 props는 재정의 하는 건 어떨까요?!

Reply:
오 이거 관련 린트 설정을 추가 해야하나 고민했는데 타입을 재선언하는 방식으로 해결할 수 있군용 !
감사합니당 ~